### PR TITLE
Adding prow config to build operator docker

### PR DESCRIPTION
**Pull request by hemantha-kumara**
Adding prow config for docker building to support xgboost operator manifests

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/xgboost-operator/37)

### CODE REVIEW

**Comment by terrytangyuan** [comment]
@hemantha-kumara Please ping me when this is ready for review.

**Comment by hemantha-kumara** [comment]
> @hemantha-kumara Please ping me when this is ready for review.

@terrytangyuan sure, will ping you once test-infra code get merges and able to build docker using presubmit.

**Comment by hemantha-kumara** [comment]
@terrytangyuan test-infra is enabled for xgboost-operator, can you please give  /ok-to-test label, so that I can start testing pre submit jobs.

**Comment by hemantha-kumara** [comment]
@terrytangyuan  @jlewi  Can you pls help me to understand this issue ?

https://prow.k8s.io/log?id=1180939420119863296&job=kubeflow-xgboost-operator-presubmit
Getting error 
`INFO|2019-10-06T20:15:04|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| level=error msg="find objects: Received status code '404' when trying to retrieve OpenAPI schema for cluster version 'v0.3.1' from URL 'https://raw.githubusercontent.com/kubernetes/kubernetes/v0.3.1/api/openapi-spec/swagger.json'"`
I dont understand how it is going to v0.3.1 version?

**Comment by johnugeorge** [comment]
Can you try changing the api version of app.yaml as in https://github.com/kubeflow/pytorch-operator/blob/master/test/workflows/app.yaml

**Comment by hemantha-kumara** [comment]
> Can you try changing the api version of app.yaml as in https://github.com/kubeflow/pytorch-operator/blob/master/test/workflows/app.yaml

@johnugeorge Thanks, after changing version presubmit job is able to proceed further.

**Comment by hemantha-kumara** [comment]
@johnugeorge @terrytangyuan now code is in reviewable condition and with these chanages operator docker is getting built.  Can you please check and let me know if any changes are required?

here is the docker built: `gcr.io/kubeflow-ci/xgboost-operator:v1.0` logs can be found in the below link
http://testing-argo.kubeflow.org/workflows/kubeflow-test-infra/kubeflow-xgboost-operator-presubmit-build-37-43e0fca-2048-fde8?tab=workflow&nodeId=kubeflow-xgboost-operator-presubmit-build-37-43e0fca-2048-fde8-2635925412&sidePanel=logs%3Akubeflow-xgboost-operator-presubmit-build-37-43e0fca-2048-fde8-2635925412%3Amain

**Comment by johnugeorge** [comment]
Do you plan to add e2e tests to the workflow?

**Comment by hemantha-kumara** [comment]
> Do you plan to add e2e tests to the workflow?

Yes, I can submit e2e test. May be I can submit later with other PR(May be once manifests changes along with testcases get merged).

**Comment by hemantha-kumara** [comment]
If it is ok, I can create an issue for e2e tests till I submit a PR(may be someone else can take if they are ready?)

**Comment by johnugeorge** [comment]
Sure. 
/lgtm
/approve

File: build_image.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+CONTEXT_DIR=$(dirname "$DOCKERFILE")
+IMAGE=$2
+TAG=$3
+
+gcloud auth activate-service-account --key-file=${GOOGLE_APPLICATION_CREDENTIALS}
+
+cd $CONTEXT_DIR
+
+echo "Building ${IMAGE} using gcloud build"
+gcloud builds submit --tag=${IMAGE}:${TAG} --project=${GCP_PROJECT} .
```
Comment (johnugeorge) [inline build_image.sh]: Can you also look at how tag is created?  https://github.com/kubeflow/pytorch-operator/blob/master/scripts/build.sh#L38 
This will ensure that unique tag is created for every presubmit

Comment (hemantha-kumara) [reply build_image.sh]: With suggested change, docker version will get changed for every submission(new docker will be created for every submission in both presubmit and postsubmit) ? Is that ok?

Comment (johnugeorge) [reply build_image.sh]: Isn't that you need? You need to have a unique image tag created for every presubmit. And you will separately add a new tag for new release. eg: v0.7 tag for upcoming release

Comment (hemantha-kumara) [reply build_image.sh]: I was thinking, manually we can update docker version only when operatror code changes instead of changing for every submission.
Automatic unique tag generation is always fine(except container registry size issue), I will resubmit as suggested.
Thanks for your comment.

Comment (hemantha-kumara) [reply build_image.sh]: @johnugeorge 
One info, currently xgboost-operator repo doesn't have any tags hence docker image tag name will be having some unique tag but it may not be in proper format like pytorch or anyother images.

till we create some branch, we may have to bear with this tags.

Comment (hemantha-kumara) [reply build_image.sh]: newly built image is `gcr.io/kubeflow-ci/xgboost-operator:de89cf4`
